Sortable fields with scope:name_of_scope

FiloSpaTeam opened the pull request:
You can sort things like this:
![image](https://github.com/guillaumebriday/jsonapi-scopes/assets/2974561/5f4b3ca4-101a-49c3-8c6e-c522c6315021)

EDIT: The array in this position seems needed to filters with dates so reverted.

FiloSpaTeam committed:
fix #2: now you can sort by scope and attributes mixed;

FiloSpaTeam committed:
convert back to string when the value is an Array; this actually fixes a lot of problems with Elasticsearch filters for example

guillaumebriday commented:
Hi! Thanks for the PR 🙌 

Can you describe what you're trying to achieve here in the PR's description? 

thank

FiloSpaTeam commented:
Hi!

Sure, first in the filters.rb i fixed a borderline situation where this code
```ruby
value = value.to_s.split(',').reject(&:blank?) if value.include?(',')
```
converts the value to an array where there are commas inside. This conflict with elasticsearch-rails gems that expects a string and never an array. But i think we should never return an array in this position, o idk what specific usage there is?

The second one is to fix the #2.
You can sort things like this:
![image](https://github.com/guillaumebriday/jsonapi-scopes/assets/2974561/5f4b3ca4-101a-49c3-8c6e-c522c6315021)

FiloSpaTeam committed:
remove join in filters that breaks date filters

FiloSpaTeam committed:
fix(filters): add safe operator for  splitting values

olistik committed:
fix(preloads): Applies `preload` instead of `includes` when the relation is polymorphic (#2)

Keeps using `allowed_includes` but applies `preload` when the relation
is polymorphic in order to overcome
`ActiveRecord::EagerLoadPolymorphicError: Cannot eagerly load the
polymorphic association xxx`.

olistik committed:
fix(includes): Skips polymorphic check on nested resources

Could be improved but had to rush to fix a regression

olistik committed:
fix(sort): Works even for empty configurations (#3)

When no `default_sort` is defined for the target class, doesn't remove
previously defined query conditions.

Before this change:

```ruby
MyModel.where(id: 3).apply_sort({})
```

`apply_sort` would remove `where(id: 3)` when `MyModel` doesn't include
the definition of `default_sort`.

This was happening because the empty behavior were returning the model
class instead of an ActiveRecord relation based on the same class,
therefore beaking the scopes chaining.

With this change, the returned object is an ActiveRecord relation even
when no `default_sort` is defined.

FiloSpaTeam committed:
fix(includes): missing includes for associations

olistik committed:
feature(allowed_includes): adds convenience `allowed_includes_list` class method

Expose allowed_includes_list on include-enabled models and cover it with
specs to verify it returns the current configured include list.

olistik committed:
Merge pull request #5 from olistik/feature/allowed_includes_list

feature(allowed_includes): adds convenience `allowed_includes_list` class method
